common/core/common.d.ts

```diff
@@ -31,9 +22,5 @@ export interface Receiver extends EventEmitter {
     on(type: string, func: Function): this;
 }
 
-export interface TransportConfig {
-    host: string;
-    deviceId: string;
-    sharedAccessSignature?: string;
-    x509?: X509;
-}
\ No newline at end of file
+export { DeviceCredentials, X509 } from './lib/authorization';
+export { DeviceCredentials as TransportConfig } from './lib/authorization';
```

Comment:
>as TransportConfig [](start = 27, length = 18)

Ouch.   I'm not a big fan of making aliases like this -- it looks sloppy and it adds cognitive overhead.  Are you planning to remove this alias?  I know we have a few different objects floating around called TransportConfig and they're almost the same but not quite.  (IIRC, one has sharedAccessSignature as a string and a different one has it as a SharedAccessSignature object.)  Conceptually, I think DeviceCredenentials should be a member of TransportConfig.   We should consider unifying all the TransportConfig objects and making DeviceCredentials a member, maybe as an MQ task.

Reply:
technically these config objects contain only device credentials, which is why I thought it was a better name. but for backward compatibility purposes, we have to keep TransportConfig. I guess I could keep that name even though it sucks a little, and we can do the renaming/cleaning later?

---
In reply to: [157019626](https://github.com/Azure/azure-iot-sdk-node/pull/179#discussion_r157019626) [](ancestors = 157019626)

Reply:
I'm fine keep this as is for now.  I just want us to remember that transport configuration could contain more than just credentials.  I think this discussion gets mixed with our setTransportOptions silliness.  At one point, before code moved around, I wanted polling interval and timeout interval as part of the transport config.

---
In reply to: [157030220](https://github.com/Azure/azure-iot-sdk-node/pull/179#discussion_r157030220) [](ancestors = 157030220,157019626)